src/encoded/static/components/experiment_series.js

```diff
@@ -192,21 +384,82 @@ class ExperimentSeriesComponent extends React.Component {
             );
         }
 
-        // Add the "Add all to cart" button and internal tags from all related datasets.
+        // Pre-processing dataset metadata
+        const seriesAssayTermName = this.props.context.assay_term_name;
+        // Some columns are assay specific, i.e. should only be shown for corresponding experiment series
+        const columns = Object.entries(experimentTableColumns).reduce(
+            (colObjs, [columnId, column]) => {
+                if (column.assay_term_name) {
+                    if (seriesAssayTermName.filter(term => column.assay_term_name.includes(term)).length > 0) {
+                        colObjs[columnId] = column;
+                    }
+                } else {
+                    colObjs[columnId] = column;
+                }
+                return colObjs;
+            },
+            {}
+        );
         let addAllToCartControl;
         let internalTags = [];
-        if (viewableDatasets.length > 0) {
+        const allRows = [];
+        const viewableDatasets = this.state.viewableDatasets;
+        if (Object.keys(viewableDatasets).length > 0) {
+            // Add the "Add all to cart" button and internal tags from all related datasets.
             addAllToCartControl = (
                 <div className="experiment-table__header">
                     <h4 className="experiment-table__title">{`Experiments in experiment series ${context.accession}`}</h4>
-                    <CartAddAllElements elements={viewableDatasets} />
+                    <CartAddAllElements elements={Object.values(viewableDatasets)} />
                 </div>
             );
 
             // Collect unique internal_tags from all relevant experiments.
-            internalTags = _.uniq(viewableDatasets.reduce((allInternalTags, experiment) => (
+            internalTags = _.uniq(Object.values(viewableDatasets).reduce((allInternalTags, experiment) => (
                 experiment.internal_tags && experiment.internal_tags.length > 0 ? allInternalTags.concat(experiment.internal_tags) : allInternalTags
             ), []));
+
+            // Experiment table data
+            Object.values(viewableDatasets).forEach((experiment) => {
```

Comment:
`Object.values` will crash on IE11. You can instead use `Object.keys(viewableDAtasets)` and then witin the `forEach` loop use `viewableDAtasets[whatever your loop variable is]` to get the corresponding value.